client/src/components/organisms/RequestGroupForm.tsx

```diff
@@ -0,0 +1,351 @@
+    }
+    if (name.length > 20) {
+      return "Group name cannot exceed 20 characters";
+    }
+    if (props.requestGroups.find((requestGroup) => requestGroup.name === name && requestGroup._id !== props.requestGroupId)) {
+      return "There is already a group with this name";
+    }
+
+    return "";
+  }
+
+  const getInputRequestTypeNameError = (inputRequestTypeName: string): string => {
+    if (requestTypeNames.find(requestTypeName => requestTypeName === inputRequestTypeName)) {
+      return "There is already a type with this name";
+    }
+    else if (inputRequestTypeName.length > 40) {
+      return "Type name cannot exceed 40 characters";
+    }
+    return "";
+  }
+
+  const getRequestTypeNamesError = (requestTypes: Array<string>): string => {
+    if (requestTypes.length === 0) {
+
+      return "Please create at least 1 type";
+    }
+    return "";
+  }
+
+  const getDescriptionError = (description: string) => {
+    if (!description) {
+      return "Please enter a description";
+    }
+
+    const editorState = EditorState.createWithContent(convertFromRaw(JSON.parse(description)))
+    if (!editorState.getCurrentContent().hasText()) {
+      return "Please enter a description";
+    }
+
+    return "";
+  }
+
+  const getImageError = (image: string) => {
+    if (image === "") {
+      return "Please select an image";
+    }
+    return "";
+  }
+
+  const onNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const newName: string = event.target.value;
+
+    setChangeMade(true);
+    setName(newName);
+    setNameError(getNameError(newName))
+  }
+
+  const onAddRequestType = (newRequestTypeName: string) => {
+    const error = getInputRequestTypeNameError(newRequestTypeName);
+
+    if (error === "") {
+      setRequestTypeNames(requestTypeNames.concat([newRequestTypeName]));
+    }
+    setChangeMade(true);
+    setRequestTypesError(error);
+  }
+
+  const onDeleteRequestType = (targetRequestTypeName: string) => {
+    const newRequestTypeNames = requestTypeNames.filter((requestTypeName) => requestTypeName !== targetRequestTypeName);
+
+    setChangeMade(true);
+    setRequestTypeNames(newRequestTypeNames);
+    setRequestTypesError(getRequestTypeNamesError(newRequestTypeNames))
+  }
+
+  const onInputRequestTypeNameChange = (requestTypeName: string) => {
+    setChangeMade(true);
+    setRequestTypesError(getInputRequestTypeNameError(requestTypeName))
+    return true;
+  }
+
+  const onDescriptionChange = (newDescription: string) => {
+    setDescription(newDescription)
+    setDescriptionError(getDescriptionError(description))
+  }
+
+  const onImageChange = (newImage: string) => {
+    setImageError(getImageError(newImage))
+    if (images.find((imageUrl) => imageUrl === newImage)) {
+      setImage(newImage)
+    }
+  }
+
+  const onSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    const tempNameError = getNameError(name);
```

Comment:
Why not just set errors right now and check the just set errors?

Reply:
setstate is async

Reply:
don't want to bother with promise chaining

Reply:
i think the setters from useState are async. also, don't want to bother with useEffect

Reply:
This is more relevant if you combine the getters with the setters as described above